rita/src/rita_common/port_emissary/mod.rs

```diff
@@ -0,0 +1,45 @@
+use std::fs::File;
+use std::io::prelude::*;
+use std::u16;
+
+/// Helper function for parsing out port number from local_address column
+fn take_local_port(s: &str) -> u16 {
+    // second column in table contains local_address
+    let local_addr = s.split_whitespace().nth(1).unwrap();
+    // having a format like "00000000:14E9"
+    let port = local_addr.split(":").nth(1).unwrap();
+
+    u16::from_str_radix(port, 16).unwrap()
+}
+
+/// Returns list of ports in use as seen in the UDP socket table (/proc/net/udp)
+fn used_ports() -> Vec<u16> {
```

Comment:
the result from take ports will probably filter up through here and then through tunnel manager until we gracefully print and drop it at the handler or Rita loop level. Crashes in Rita are a very big deal as we don't often have a large amount of environmental control or an easy way to restart Rita remotely. (actually a cron job to restart rita all the time if it's not running might be a good use of time...)